sound/soc/sof/topology.c

```diff
@@ -1890,9 +1890,7 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 	if (ret != 0) {
 		dev_err(sdev->dev, "error: parse process.cfg tokens failed %d\n",
 			le32_to_cpu(private->size));
-		kfree(wdata);
-		kfree(process);
-		return ret;
+		goto err;
```

Comment:
one doubt (mostly paranoia since we were burned with the MSI stuff on a return value).
This is different to all other tests, where ret < 0, can you double-check that there is no case where we return a positive value and hence end-up not doing the kfree(process)

Reply:
@plbossart this check is fine. sof_parse_tokens() only returns something < 0 on error or 0 upon success. 
The other checks are < 0 because the error in the reply message cannot be guaranteed to be only 0 on success.

Reply:
ok then let's merge. I'll squash it as v2 for upstream